documentation/src/main/asciidoc/topics/proc_configuring_textbased_encoding.adoc

```diff
@@ -0,0 +1,33 @@
+[id='using_text_based_storage-{context}']
+= Using Text-Based Formats for Data
+You can configure {brandname} to encode data in text-based formats such as
+`text/plain`, `application/json`, or `application/xml`.
+
+.Procedure
+
+. Specify a text-based storage format as the MediaType for keys and values.
+. Optionally specify a character set such as `UTF-8`.
++
+The following example configures {brandname} to store entries with the `text/plain; charset=UTF-8` format:
++
+[source,xml,options="nowrap",subs=attributes+]
+----
+include::config_examples/encoding_text_plain.xml[]
+----
++
+. Configure your clients.
+
+Hot Rod clients can use `org.infinispan.commons.marshall.StringMarshaller`
+to handle plain text, JSON, XML, or any other text-based format.
+
+If you want to use text-based formats and binary data with Hot Rod clients, you
+should use ProtoStream marshalling. ProtoStream can handle plain text and JSON 
+without the need to create Serialization Contexts and register Protobuf
+schemas. However, ProtoStream does not support XML.
+
+ProtoStream offers more flexibility for storing data in binary format. However, you must  and provide Protobuf schemas (`.proto` files) to use ProtoStream.
```

Comment:
I sense some duality between "text" and "binary" throughout the docs. 

Let me clarify: "text" is also binary for all it matters. ```text/plain```, ```application/json```, ```application/xml```, are all stored as ```byte[]``` inside the cache. The media type is just a description of what is inside that ```byte[]```.

Reply:
Also, I think we haven't been camel casing ```Protostream```

Reply:
hmm. I've used camel case for ProtoStream elsewhere in docs. it also appears that way at: https://javadoc.io/doc/org.infinispan.protostream/protostream/latest/index.html

Reply:
Sorry, I was wrong. Even ProtoStream itself uses it :smile: